### ISSUE TITLE
Issue with Open Sans font in social plugin

### ISSUE THREAD

**Issue by croesus** [squidfunk/mkdocs-material#5517]
### Bug description

Google's Open Sans font package contains several font variations in subdirectories. The code picks the last one that ends in "-Regular.ttf" which returns an arbitrary variation based on the order the OS returns files.

```
Open_Sans
├── OFL.txt
├── OpenSans-Italic-VariableFont_wdth,wght.ttf
├── OpenSans-VariableFont_wdth,wght.ttf
├── README.txt
└── static
    ├── OpenSans
    │   ├── OpenSans-Bold.ttf
    │   ├── OpenSans-BoldItalic.ttf
    │   ├── OpenSans-ExtraBold.ttf
    │   ├── OpenSans-ExtraBoldItalic.ttf
    │   ├── OpenSans-Italic.ttf
    │   ├── OpenSans-Light.ttf
    │   ├── OpenSans-LightItalic.ttf
    │   ├── OpenSans-Medium.ttf
    │   ├── OpenSans-MediumItalic.ttf
    │   ├── OpenSans-Regular.ttf
    │   ├── OpenSans-SemiBold.ttf
    │   └── OpenSans-SemiBoldItalic.ttf
    ├── OpenSans_Condensed
    │   ├── OpenSans_Condensed-Bold.ttf
    │   ├── OpenSans_Condensed-BoldItalic.ttf
    │   ├── OpenSans_Condensed-ExtraBold.ttf
    │   ├── OpenSans_Condensed-ExtraBoldItalic.ttf
    │   ├── OpenSans_Condensed-Italic.ttf
    │   ├── OpenSans_Condensed-Light.ttf
    │   ├── OpenSans_Condensed-LightItalic.ttf
    │   ├── OpenSans_Condensed-Medium.ttf
    │   ├── OpenSans_Condensed-MediumItalic.ttf
    │   ├── OpenSans_Condensed-Regular.ttf
    │   ├── OpenSans_Condensed-SemiBold.ttf
    │   └── OpenSans_Condensed-SemiBoldItalic.ttf
    └── OpenSans_SemiCondensed
        ├── OpenSans_SemiCondensed-Bold.ttf
        ├── OpenSans_SemiCondensed-BoldItalic.ttf
        ├── OpenSans_SemiCondensed-ExtraBold.ttf
        ├── OpenSans_SemiCondensed-ExtraBoldItalic.ttf
        ├── OpenSans_SemiCondensed-Italic.ttf
        ├── OpenSans_SemiCondensed-Light.ttf
        ├── OpenSans_SemiCondensed-LightItalic.ttf
        ├── OpenSans_SemiCondensed-Medium.ttf
        ├── OpenSans_SemiCondensed-MediumItalic.ttf
        ├── OpenSans_SemiCondensed-Regular.ttf
        ├── OpenSans_SemiCondensed-SemiBold.ttf
        └── OpenSans_SemiCondensed-SemiBoldItalic.ttf
```
I've implemented a change to the social plugin that specifically matches the <FontName>-Regular.ttf version in [this fork](https://github.com/icebreakerone/mkdocs-material/blob/master/src/plugins/social/plugin.py). It uses `os.walk()` rather than `glob()` to find the files. Either would probably be fine but I'd implemented it before your fix for the Inter font (#5420 - see related links).

It sounds like work is in progress on the social plugin, so please let me know if you would like me to submit a PR for this or just pick it up yourself.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [TypeError: expected string or bytes-like object in social plugin #5420](https://github.com/squidfunk/mkdocs-material/issues/5420)

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/11477123/example.zip)

### Steps to reproduce

If you add 
```
                print(f"{match.group(1)} - {font[match.group(1)]}")
```
on line 465 of plugin.py then run `mkdocs build` you get the output
```
...
Regular - .cache/plugin/social/static/OpenSans_Condensed/OpenSans_Condensed-Regular.ttf
...
Regular - .cache/plugin/social/static/OpenSans_SemiCondensed/OpenSans_SemiCondensed-Regular.ttf
...
Regular - .cache/plugin/social/static/OpenSans/OpenSans-Regular.ttf
...
```
It's an accident of file ordering that the last one is the one we want, and [shouldn't be relied upon](https://stackoverflow.com/questions/6773584/how-are-glob-globs-return-values-ordered). 

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by croesus** [comment]
Thanks for the quick response. [PR submitted](https://github.com/squidfunk/mkdocs-material/pull/5518).

**Comment by squidfunk** [comment]
Resolved by #5518.

**Comment by squidfunk** [comment]
Released as part of [9.1.13](https://squidfunk.github.io/mkdocs-material/changelog/#9.1.13).
